src/libs/pddl_parser/tests/test_pddl_parser.cpp

```diff
@@ -0,0 +1,261 @@
+	 )                                          \n\
+	(:action test-action                        \n\
+	 :parameters (?t - obj-a)                   \n\
+	 :precondition (pred ?t)                    \n\
+	 :effect (not (pred ?t))                    \n\
+	 )                                          \n\
+)",
+	                   "(define (domain action-unknown-type-in-param)    \n\
+	(:requirements :strips :typing)             \n\
+	(:types                                     \n\
+	 	obj-a - object                            \n\
+	 )                                          \n\
+	(:predicates                                \n\
+	 	(pred ?r - obj-a)                         \n\
+	 )                                          \n\
+	(:action test-action                        \n\
+	 :parameters (?t - obj-b)                   \n\
+	 :precondition (pred ?t)                    \n\
+	 :effect (not (pred ?t))                    \n\
+	 )                                          \n\
+)",
+	                   "(define (domain action-unknown-type-in-constant) \n\
+	(:requirements :strips :typing)             \n\
+	(:types                                     \n\
+	 	obj-a - object                            \n\
+	 )                                          \n\
+	(:constants                                 \n\
+	 	TEST_CONST - obj-b                        \n\
+	 )                                          \n\
+	(:predicates                                \n\
+	 	(pred ?r - obj-a)                         \n\
+	 )                                          \n\
+	(:action test-action                        \n\
+	 :parameters (?t - obj-a)                   \n\
+	 :precondition (pred ?t)                    \n\
+	 :effect (not (pred ?t))                    \n\
+	 )                                          \n\
+)",
+	                   "(define (domain action-constant-missmatch)       \n\
+	(:requirements :strips :typing)             \n\
+	(:types                                     \n\
+	 	obj-a - object                            \n\
+	 	obj-b - object                            \n\
+	 )                                          \n\
+	(:constants                                 \n\
+	 	TEST_CONST - obj-b                        \n\
+	 )                                          \n\
+	(:predicates                                \n\
+	 	(pred ?r - obj-a)                         \n\
+	 )                                          \n\
+	(:action test-action                        \n\
+	 :parameters (?t - obj-a)                   \n\
+	 :precondition (pred TEST_CONST)            \n\
+	 :effect (not (pred ?t))                    \n\
+	 )                                          \n\
+)"};
+	for (const auto &s : benchmarks) {
+		try {
+			PddlParser p;
+			p.parseDomain(s);
+			FAIL() << s << "\n expected type error";
+		} catch (PddlParserException &e) {
+			if (e.error_type != PddlErrorType::TYPE_ERROR) {
+				FAIL() << s << "\n failed with error " << e.what();
+			} else {
+			}
```

Comment:
see above, you could check for the correct exception type directly

Also, this fails if no exception is thrown at all (as you will never reach the catch block)

Reply:
Additionally, you don't test whether the parsed content actually makes sense. Is this too complicated?

Reply:
Simplified the tests in https://github.com/fawkesrobotics/fawkes/pull/292/commits/de25ea6317bf114b0fc06e6d752d1ca3ab76f156

Also attempted to check whether parsed content makes sense in https://github.com/fawkesrobotics/fawkes/pull/292/commits/892e3f0299103273999fe06e033006d943faf565.

Not pretty but it does what it should (and i actually found and fixed a bug with it in  https://github.com/fawkesrobotics/fawkes/pull/292/commits/62af660af6fe89006bbd0a2452e416c8e825d938)